netdefine.go

```diff
@@ -177,126 +275,167 @@ func Create(cfg *Config) error {
 	for _, p := range cfg.Peers {
 		_, ok := peers[p.Name]
 		if ok {
-			return fmt.Errorf("duplicate peer name: %s", p.Name)
+			return nil, fmt.Errorf("duplicate peer name: %s", p.Name)
 		}
 		peers[p.Name] = true
 
 		for net, l := range p.Links {
 			if _, ok := nets[net]; !ok {
-				return fmt.Errorf("peer %s has link to non-existent network %q", p.Name, net)
+				return nil, fmt.Errorf("peer %s has link to non-existent network %q", p.Name, net)
 			}
 
 			if err := l.Parse(); err != nil {
-				return err
+				return nil, err
+			}
+		}
+	}
+
+	for name, net := range nets {
+		for targetNet := range net.Links {
+			if _, ok := nets[targetNet]; !ok {
```

Comment:
one bit of probably helpful UX:

We could check if the name they gave was a peer, and give a special error message saying you can't link a network to a peer.